<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from ./ur5_robot_with_gripper.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="ur5" 

xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller" xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface" 
xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor" 
xmlns:xacro="http://ros.org/wiki/xacro">
 
<mujoco>
	<compiler meshdir="/home/student/mjpro140/model/ur5/meshes" 
		  discardvisual = "true"
		  balanceinertia="true" 
		  boundinertia="0.00001" 
		  boundmass="0.001"
		  inertiafromgeom="true" 
		  angle="radian"/>

	<option   timestep="0.01" gravity="0 0 -9.8" viscosity="0" />
</mujoco>
 

<!-- common stuff -->
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="ros_control">
      <!--robotNamespace>/</robotNamespace-->
      <!--robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType-->
    </plugin>
    <!--
    <plugin name="gazebo_ros_power_monitor_controller" filename="libgazebo_ros_power_monitor.so">
      <alwaysOn>true</alwaysOn>
      <updateRate>1.0</updateRate>
      <timeout>5</timeout>
      <powerStateTopic>power_state</powerStateTopic>
      <powerStateRate>10.0</powerStateRate>
      <fullChargeCapacity>87.78</fullChargeCapacity>     
      <dischargeRate>-474</dischargeRate>
      <chargeRate>525</chargeRate>
      <dischargeVoltage>15.52</dischargeVoltage>
      <chargeVoltage>16.41</chargeVoltage>
    </plugin>-->
  </gazebo>

<!-- ur5 -->
  <material name="DarkGrey">
    <color rgba="0.3 0.3 0.3 1.0"/>
  </material>
  <material name="DarkerGrey">
    <color rgba="0.15 0.15 0.15 1.0"/>
  </material>
  <material name="Black">
    <color rgba="0 0 0 1.0"/>
  </material>
  <material name="Orange">
    <color rgba="1.0 0.487 0 1.0"/>
  </material>
  <material name="UniHB-Red">
    <!--UniBremen Red is Verkehrsrot / RAL 7035 ~= RGB 193 18 28 -->
    <color rgba=".7568627 .070588 .10980392 1.0"/>
  </material>
  <material name="Silver">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <material name="Grey">
    <color rgba="0.7 0.7 0.7 1.0"/>
  </material>
  <material name="Blue">
    <color rgba="0.0 0.0 0.8 1.0"/>
  </material>
  <material name="HandGray">
    <color rgba="0.953 0.996 0.694 1.0"/>
  </material>
  <material name="SickBlue">
    <color rgba="0.3058 0.5921 0.7294 1.0"/>
  </material>
  <!-- <xacro:include filename="$(find iai_kuka_lwr4_description)/urdf/kuka_lwr_arm.urdf.xacro"/> -->
  <!--xacro:include filename="$(find iai_boxy_description)/sensors/kinect.urdf.xacro"/-->
  <!-- <xacro:include filename="$(find iai_adapters_description)/urdf/adapter_boxy_ur3.urdf.xacro"/> -->
  <!-- <xacro:include filename="$(find iai_ur_description)/urdf/ur3.urdf.xacro" /> -->
  <!-- <xacro:include filename="$(find iai_kinect2_description)/urdf/kinect2.urdf.xacro"/> -->
  <!-- Inertia parameters -->
  <!-- This mass might be incorrect -->
  <!-- Kinematic model -->
  <!-- Properties from urcontrol.conf -->
  <!--
    DH for UR5:
    a = [0.00000, -0.42500, -0.39225,  0.00000,  0.00000,  0.0000]
    d = [0.089159,  0.00000,  0.00000,  0.10915,  0.09465,  0.0823]
    alpha = [ 1.570796327, 0, 0, 1.570796327, -1.570796327, 0 ]
    q_home_offset = [0, -1.570796327, 0, -1.570796327, 0, 0]
    joint_direction = [-1, -1, 1, 1, 1, 1]
    mass = [3.7000, 8.3930, 2.2750, 1.2190, 1.2190, 0.1879]
    center_of_mass = [ [0, -0.02561, 0.00193], [0.2125, 0, 0.11336], [0.11993, 0.0, 0.0265], [0, -0.0018, 0.01634], [0, 0.0018,0.01634], [0, 0, -0.001159] ]
  -->
  <!-- Arbitrary offsets for shoulder/elbow joints -->
  <!-- measured from model -->
  <!-- measured from model -->
  <!-- link lengths used in model -->
  <!--property name="shoulder_height" value="0.089159" /-->
  <!--property name="shoulder_offset" value="0.13585" /-->
  <!-- shoulder_offset - elbow_offset + wrist_1_length = 0.10915 -->
  <!--property name="upper_arm_length" value="0.42500" /-->
  <!--property name="elbow_offset" value="0.1197" /-->
  <!-- CAD measured -->
  <!--property name="forearm_length" value="0.39225" /-->
  <!--property name="wrist_1_length" value="0.093" /-->
  <!-- CAD measured -->
  <!--property name="wrist_2_length" value="0.09465" /-->
  <!-- In CAD this distance is 0.930, but in the spec it is 0.09465 -->
  <!--property name="wrist_3_length" value="0.0823" /-->
  <!-- manually measured -->
  <!-- manually measured -->
  <!-- manually measured -->
  <!-- manually measured -->
  <!-- manually measured -->
 
<!-- arm -->
  <link name="base_link">
    <visual>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/Base.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/base.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="4.0"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.00443333156" ixy="0.0" ixz="0.0" iyy="0.00443333156" iyz="0.0" izz="0.0072"/>
    </inertial>
  </link>
  <joint name="shoulder_pan_joint" type="revolute">
    <parent link="base_link"/>
    <child link="shoulder_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.089159"/>
    <axis xyz="0 0 1"/>
    <limit effort="150.0" lower="-3.14159265" upper="3.14159265" velocity="3.15"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="shoulder_link">
    <visual>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/Shoulder.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/shoulder.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="3.7"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.010267495893" ixy="0.0" ixz="0.0" iyy="0.010267495893" iyz="0.0" izz="0.00666"/>
    </inertial>
  </link>
  <joint name="shoulder_lift_joint" type="revolute">
    <parent link="shoulder_link"/>
    <child link="upper_arm_link"/>
    <origin rpy="0.0 1.570796325 0.0" xyz="0.0 0.13585 0.0"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-3.14159265" upper="3.14159265" velocity="3.15"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="upper_arm_link">
    <visual>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/UpperArm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/upper_arm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="8.393"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.28"/>
      <inertia ixx="0.22689067591" ixy="0.0" ixz="0.0" iyy="0.22689067591" iyz="0.0" izz="0.0151074"/>
    </inertial>
  </link>
  <joint name="elbow_joint" type="revolute">
    <parent link="upper_arm_link"/>
    <child link="forearm_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.1197 0.425"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-3.14159265" upper="3.14159265" velocity="3.15"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="forearm_link">
    <visual>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/Forearm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/forearm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="2.275"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.25"/>
      <inertia ixx="0.049443313556" ixy="0.0" ixz="0.0" iyy="0.049443313556" iyz="0.0" izz="0.004095"/>
    </inertial>
  </link>
  <joint name="wrist_1_joint" type="revolute">
    <parent link="forearm_link"/>
    <child link="wrist_1_link"/>
    <origin rpy="0.0 1.570796325 0.0" xyz="0.0 0.0 0.39225"/>
    <axis xyz="0 1 0"/>
    <limit effort="28.0" lower="-3.14159265" upper="3.14159265" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_1_link">
    <visual>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/Wrist1.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/wrist_1.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.219"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.111172755531" ixy="0.0" ixz="0.0" iyy="0.111172755531" iyz="0.0" izz="0.21942"/>
    </inertial>
  </link>
  <joint name="wrist_2_joint" type="revolute">
    <parent link="wrist_1_link"/>
    <child link="wrist_2_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.093 0.0"/>
    <axis xyz="0 0 1"/>
    <limit effort="28.0" lower="-3.14159265" upper="3.14159265" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_2_link">
    <visual>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/Wrist2.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/wrist_2.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.219"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.111172755531" ixy="0.0" ixz="0.0" iyy="0.111172755531" iyz="0.0" izz="0.21942"/>
    </inertial>
  </link>
  <joint name="wrist_3_joint" type="revolute">
    <parent link="wrist_2_link"/>
    <child link="wrist_3_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.09465"/>
    <axis xyz="0 1 0"/>
    <limit effort="28.0" lower="-3.14159265" upper="3.14159265" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="wrist_3_link">
    <visual>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/Wrist3.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/wrist_3.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.1879"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0171364731454" ixy="0.0" ixz="0.0" iyy="0.0171364731454" iyz="0.0" izz="0.033822"/>
    </inertial>
  </link>
  <joint name="ee_fixed_joint" type="fixed">
    <parent link="wrist_3_link"/>
    <child link="ee_link"/>
    <origin rpy="-1.570796325 0 0" xyz="0.0 0.0823 0.0"/>
  </joint>
  <link name="ee_link">
    <collision>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.01 0 0"/>
    </collision>
  </link>
  <!-- The following URDFs have frame_in / frame_out, so they are simply placed at their parent -->
  <!-- Joint connecting the parent to our frame_in -->
  <joint name="adapter_iso50_kms40_joint" type="fixed">
    <origin rpy="0 0 4.712388975" xyz="0 0 0"/>
    <parent link="ee_link"/>
    <child link="adapter_iso50_kms40_frame_in"/>
  </joint>
  <!-- link for the adapter plate -->
  <link name="adapter_iso50_kms40_frame_in">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/adapter_plate_iso50_kms40_weiss.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="Grey"/>
    </visual>
  </link>
  <!-- Empty link just to get the frame out -->
  <link name="adapter_iso50_kms40_frame_out"/>
  <!-- joint positioning the frame_out -->
  <joint name="adapter_iso50_kms40_trans_frame_out" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.016"/>
    <parent link="adapter_iso50_kms40_frame_in"/>
    <child link="adapter_iso50_kms40_frame_out"/>
    <axis xyz="0 0 0"/>
  </joint>
  <!-- Now joint connecting the actual kms40 sensor -->
  <joint name="kms40_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="adapter_iso50_kms40_frame_out"/>
    <child link="kms40_frame_in"/>
  </joint>
  <!-- Link for the kms40 sensor-->
  <link name="kms40_frame_in">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/kms40_simple.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="Grey"/>
    </visual>
  </link>
  <!-- Empty link just to get the frame out -->
  <!-- This is also the reference frame for the force/torque data -->
  <link name="kms40_frame_out"/>
  <!-- joint positioning the frame_out -->
  <joint name="kms40_trans_frame_out" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.027"/>
    <parent link="kms40_frame_in"/>
    <child link="kms40_frame_out"/>
    <axis xyz="0 0 0"/>
  </joint>
  <!-- Joint connecting the parent to our frame_in -->
  <joint name="adapter_kms40_fwk050_in_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="kms40_frame_out"/>
    <child link="adapter_kms40_fwk050_frame_in"/>
  </joint>
  <link name="adapter_kms40_fwk050_frame_in">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!-- these meshes were exported in mm, so scale back to m -->
        <mesh filename="package://mujoco_test/models/meshes/adapter_kms40_fwk050.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="Grey"/>
    </visual>
  </link>
  <!-- Empty link just to get the frame out -->
  <link name="adapter_kms40_fwk050_frame_out"/>
  <!-- joint positioning the frame_out -->
  <joint name="adapter_kms40_fwk050_trans_frame_out" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.01"/>
    <parent link="adapter_kms40_fwk050_frame_in"/>
    <child link="adapter_kms40_fwk050_frame_out"/>
    <axis xyz="0 0 0"/>
  </joint>
  <!-- The joint connecting us to the next part of the chain is in the next URDF -->
  <!-- Joint connecting the parent to our frame_in -->
  <joint name="fwk_fwa_050_in_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="adapter_kms40_fwk050_frame_out"/>
    <child link="fwk_fwa_050_frame_in"/>
  </joint>
  <link name="fwk_fwa_050_frame_in">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!-- these meshes were exported in mm, so scale back to m -->
        <mesh filename="package://mujoco_test/models/meshes/fwk_fwa_050_simple.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="Grey"/>
    </visual>
  </link>
  <!-- Empty link just to get the frame out -->
  <link name="fwk_fwa_050_frame_out"/>
  <!-- joint positioning the frame_out -->
  <joint name="fwk_fwa_050_trans_frame_out" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.0141"/>
    <parent link="fwk_fwa_050_frame_in"/>
    <child link="fwk_fwa_050_frame_out"/>
    <axis xyz="0 0 0"/>
  </joint>
  <!-- The joint connecting us to the next part of the chain is in the next URDF -->
  <!-- Joint connecting the parent to our frame_in -->
  <joint name="adapter_fwa050_wsg50_in_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.0"/>
    <parent link="fwk_fwa_050_frame_out"/>
    <child link="adapter_fwa050_wsg50_frame_in"/>
  </joint>
  <link name="adapter_fwa050_wsg50_frame_in">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!-- these meshes were exported in mm, so scale back to m -->
        <mesh filename="package://mujoco_test/models/meshes/adapter_fwa050_wsg50.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="Grey"/>
    </visual>
  </link>
  <!-- Empty link just to get the frame out -->
  <link name="adapter_fwa050_wsg50_frame_out"/>

  <!-- joint positioning the frame_out -->
  <joint name="adapter_fwa050_wsg50_trans_frame_out" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.01"/>
    <parent link="adapter_fwa050_wsg50_frame_in"/>
    <child link="adapter_fwa050_wsg50_frame_out"/>
    <axis xyz="0 0 0"/>
  </joint>
  <!-- The joint connecting us to the next part of the chain is in the next URDF -->

  <joint name="gripper_anterior_gripper_joint" type="fixed">
    <origin rpy="0 0 3.14159265359" xyz="0 0 0"/>
    <parent link="adapter_fwa050_wsg50_frame_out"/>
    <child link="gripper_base_link"/>
  </joint>

  <!-- BASE LINK -->
  <link name="gripper_base_link">
    <inertial>
      <mass value="0.25"/>
      <origin xyz="0 0 0"/>
      <!--inertia ixx="1.0" ixy="0.0" ixz="0.0"
               iyy="100.0" iyz="0.0" 
               izz="1.0" /-->
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/WSG50_110.stl" scale="1 1 1"/>
        <!--box size="0.146 0.05 0.0735"/-->
      </geometry>
      <material name="grey">
        <color rgba="0.5 0.5 0.5 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0 0 0"/>
      <geometry>
        <!--box size="0.146 0.05 0.0735"/-->
        <mesh filename="package://mujoco_test/models/meshes/WSG50_110.stl" scale="1 1 1"/>
      </geometry>
    </collision>
  </link>
  <gazebo reference="gripper_base_link">
    <material>Gazebo/Grey</material>
    <turnGravityOff>false</turnGravityOff>
  </gazebo>
  <!-- GRIPPER LEFT -->
  <joint name="gripper_base_left_finger_joint" type="prismatic">
    <limit effort="1.0" lower="-0.055" upper="-0.0027" velocity="0.42"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <!--origin xyz="-0.0067 0 0.049" rpy="0 0 0" /-->
    <parent link="gripper_base_link"/>
    <child link="gripper_gripper_left_link"/>
    <dynamics damping="100" friction="100"/>
    <axis xyz="1 0 0"/>
    <!-- <limit effort="100" velocity="100"/> -->
  </joint>
  <transmission name="gripper_base_trans_left" type="pr2_mechanism_model/SimpleTransmission">
    <actuator name="gripper_base_motor_left"/>
    <joint name="gripper_base_gripper_left_joint"/>
    <mechanicalReduction>1</mechanicalReduction>
    <motorTorqueConstant>1</motorTorqueConstant>
  </transmission>
  <link name="gripper_gripper_left_link">
    <inertial>
      <mass value="0.1"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/GUIDE_WSG50_110.stl" scale="0.001 0.001 0.001"/>
        <!--box size="0.0175 0.03 0.024"/-->
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!--box size="0.0175 0.03 0.024"/-->
        <mesh filename="package://mujoco_test/models/meshes/GUIDE_WSG50_110.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
    <contact_coefficients kd="1.0" kp="1000.0" mu="0"/>
  </link>
  <gazebo reference="gripper_gripper_left_link">
    <material>Gazebo/Blue</material>
    <turnGravityOff>false</turnGravityOff>
  </gazebo>
  <!-- LEFT FINGER -->
  <joint name="gripper_guide_joint_finger_left" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.023"/>
    <parent link="gripper_gripper_left_link"/>
    <child link="gripper_finger_left_link"/>
    <dynamics damping="100" friction="100"/>
    <axis xyz="1 0 0"/>
    <!-- <limit effort="100" velocity="100"/> -->
  </joint>
  <link name="gripper_finger_left_link">
    <inertial>
      <mass value="0.1"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/WSG-FMF.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/WSG-FMF.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
    <contact_coefficients kd="1.0" kp="1000.0" mu="0"/>
  </link>
  <gazebo reference="gripper_finger_left">
    <material>Gazebo/Blue</material>
    <turnGravityOff>false</turnGravityOff>
  </gazebo>
  <!-- GRIPPER RIGHT -->
  <joint name="gripper_base_right_finger_joint" type="prismatic">
    <limit effort="1.0" lower="0.0027" upper="0.055" velocity="0.42"/>
    <origin rpy="0 0 3.14159" xyz="0 0 0"/>
    <parent link="gripper_base_link"/>
    <child link="gripper_gripper_right_link"/>
    <axis xyz="-1 0 0"/>
    <dynamics damping="100" friction="100"/>
    <!-- <limit effort="100" velocity="100"/>-->
  </joint>
  <transmission name="gripper_base_trans_right" type="pr2_mechanism_model/SimpleTransmission">
    <actuator name="gripper_base_motor_right"/>
    <joint name="gripper_base_gripper_right_joint"/>
    <mechanicalReduction>1</mechanicalReduction>
    <motorTorqueConstant>1</motorTorqueConstant>
  </transmission>
  <link name="gripper_gripper_right_link">
    <inertial>
      <mass value="0.1"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/GUIDE_WSG50_110.stl" scale="0.001 0.001 0.001"/>
        <!--box size="0.0175 0.03 0.024"/-->
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <!--box size="0.0175 0.03 0.024"/-->
        <mesh filename="package://mujoco_test/models/meshes/GUIDE_WSG50_110.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
    <contact_coefficients kd="1.0" kp="1000.0" mu="0"/>
  </link>
  <gazebo reference="gripper_right">
    <material>Gazebo/White</material>
    <turnGravityOff>false</turnGravityOff>
  </gazebo>
  <!-- RIGHT FINGER -->
  <joint name="gripper_guide_joint_finger_right" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.023"/>
    <parent link="gripper_gripper_right_link"/>
    <child link="gripper_finger_right_link"/>
  </joint>
  <link name="gripper_finger_right_link">
    <inertial>
      <mass value="0.1"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/WSG-FMF.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_test/models/meshes/WSG-FMF.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
    <contact_coefficients kd="1.0" kp="1000.0" mu="0"/>
  </link>
  <gazebo reference="gripper_finger_right">
    <material>Gazebo/White</material>
    <turnGravityOff>false</turnGravityOff>
  </gazebo>
  <link name="gripper_tool_frame"/>
  <joint name="gripper_tool_frame_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.154"/>
    <parent link="gripper_base_link"/>
    <child link="gripper_tool_frame"/>
  </joint>
  <transmission name="shoulder_pan_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_pan_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_pan_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="shoulder_lift_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_lift_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_lift_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="elbow_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="elbow_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="elbow_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_1_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_2_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_3_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_3_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_3_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <!-- nothing to do here at the moment -->
  <link name="world"/>
  <joint name="world_joint" type="fixed">
    <parent link="world"/>
    <child link="base_link"/>
    <origin rpy="0.0 0.0 3.14" xyz="0.0 0.0 0.0"/>
  </joint>
</robot>

